### PR TITLE
EPUB galley article files recognized as EPUB files

### PR DESCRIPTION
**Pull request by felixgruender**
When an EPUB file is uploaded, the correct mime type (application/epub+zip) is set.

### CODE REVIEW

File: classes/file/ArticleFileManager.inc.php [hunk 1]
```diff
@@ -509,6 +509,8 @@ function handleUpload($fileName, $fileStage, $fileId = null, $overwrite = false)
 
 		if (preg_match('/\.css$/', $_FILES[$fileName]['name'])) {
 			$articleFile->setFileType('text/css');
+		} else if (preg_match('/\.epub$/', $_FILES[$fileName]['name'])) {
+			$articleFile->setFileType('application/epub+zip');
 		} else {
 			$articleFile->setFileType($this->getUploadedFileType($fileName));
```
Comment (ctgraham) [inline classes/file/ArticleFileManager.inc.php]: As of 2.4.5, getUploadedFileType() now performs enhanced file extension checking which makes the above checks (existing text/css and new application/epub+zip) undesirable.

If MIME detection typically identifies an epub as application/zip, modify the String::mime_content_type() ambiguities to override: https://github.com/pkp/pkp-lib/blob/b0005fead5591a491e3034ef4debe3634da35ae3/classes/core/String.inc.php#L409

For extra credit, move these ambiguities to the database and create an interface for the Site Admin or Journal Manager to edit them.  I should have done that in the first place.

File: classes/submission/form/ArticleGalleyForm.inc.php [hunk 2]
```diff
@@ -213,6 +213,9 @@ function execute($fileName = null, $createRemote = false) {
 			// Upload galley file
 			if ($articleFileManager->uploadedFileExists($fileName)) {
 				$fileType = $articleFileManager->getUploadedFileType($fileName);
+				if (preg_match('/\.epub$/', $_FILES[$fileName]['name'])) {
```
Comment (ctgraham) [inline classes/submission/form/ArticleGalleyForm.inc.php]: The above comment about String::mime_content_type() applies here as well.

**Comment by felixgruender** [comment]
Thanks for the hints - the superfluous file type checks for css and epub were removed. Instead of that new mime type 'epub' was added in String::mime_content_type - see the pull request in pkp/lib)

**Comment by asmecher** [comment]
Thanks, Felix -- Clinton, can I leave this to you to review based on the above?

**Comment by ctgraham** [comment]
I tested this locally and the merge of this and https://github.com/pkp/pkp-lib/pull/358 look good to me.

I'm not following why the Travis CI build failed with:
```
1) ParscitRawCitationNlm30CitationSchemaFilterTest::testExecute

Error in test #0: 'Sheril, R. D. (1956). The terrifying future: Contemplating color television. San Diego: Halstead.'.

Failed asserting that null is not null.

/home/travis/build/pkp/ojs/lib/pkp/tests/plugins/metadata/nlm30/filter/Nlm30CitationSchemaFilterTestCase.inc.php:65

/home/travis/build/pkp/ojs/lib/pkp/tests/plugins/citationParser/parscit/filter/ParscitRawCitationNlm30CitationSchemaFilterTest.php:64
``` 

Any hints from you, @asmecher ?

**Comment by asmecher** [comment]
@ctgraham, I suspect it was a temporary server problem with Parscit or similar. Certainly nothing caused by this change. Thanks, Clinton & Felix!

**Comment by asmecher** [comment]
Actually, looking again -- this reverts http://pkp.sfu.ca/bugzilla/show_bug.cgi?id=8607 which I don't think is intentional. That CSS hack was specifically added because MIME type detection for CSS files is often flaky.

**Comment by ctgraham** [comment]
No, that change is intentional.  String::mime_content_type() really should cover this as of https://github.com/pkp/pkp-lib/pull/165 .

**Comment by asmecher** [comment]
Ah, thanks -- I missed that.

**Pull request merged**